pandas/core/indexes/timedeltas.py

```diff
@@ -829,7 +829,8 @@ def _maybe_cast_slice_bound(self, label, side, kind):
             else:
                 return (lbound + to_offset(parsed.resolution) -
                         Timedelta(1, 'ns'))
-        elif is_integer(label) or is_float(label):
+        elif ((is_integer(label) or is_float(label)) and
+              not is_timedelta64_dtype(label)):
```

Comment:
Would using something like `is_number(label) and not is_timedelta64_dtype(label)` be more appropriate here?

Right now it looks like a few things can sneak through, e.g. booleans:
```python
In [2]: s = pd.Series(list('abcde'), pd.timedelta_range(0, 4, freq='ns'))

In [3]: s
Out[3]:
00:00:00           a
00:00:00.000000    b
00:00:00.000000    c
00:00:00.000000    d
00:00:00.000000    e
Freq: N, dtype: object

In [4]: s.loc[False:True]
Out[4]:
00:00:00           a
00:00:00.000000    b
Freq: N, dtype: object
```
This doesn't seem like the intended behavior, and `is_number` returns `True` for booleans.

Reply:
``is_number`` is pretty general, technically a ``bool`` is a number (as it derives from int, as does np.timedelta).

Reply:
> `is_number` is pretty general

Yes, this was the point I was trying to make. The current approach here doesn't look general enough, as things like booleans are still allowed, as per my example.  Seems like using `is_number(label) and not is_timedelta64_dtype(label)` would catch this appropriately.

Reply:
Yes, this is a gotcha. See https://github.com/numpy/numpy/issues/10685 for the upstream numpy issue.

Reply:
@jschendel you make a good point

can u open an issue (or PR!)

Reply:
@jreback : Will do.  Looking into this now more generally across various types of indexes.